#!/usr/bin/env python
PACKAGE = "daa_color_classification"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_V",                   int_t,    0, "minimum valid value (V)", 30,  0, 60)
gen.add("max_V",                   int_t,    0, "maximum valid value (V)", 255,  200, 255)
gen.add("min_S",                   int_t,    0, "minimum valid saturation (S)", 128,  0, 255)

gen.add("upper_black_V",           int_t,    0, "upper threshold of V for filtering black pixels", 10,  1, 20)

gen.add("hBins",                   int_t,    1, "H bin size for the histogram", 32, 30, 255)
gen.add("sBins",                   int_t,    0, "S bin size for the histogram", 32, 1, 255)
gen.add("min_hits_per_bin",        int_t,    0, "minimum hits per bin (will truncate to 0 below this number)", 3,  0, 1000)
gen.add("max_hits_per_bin",        int_t,    0, "maximum hits per bin (will truncate beyond this number)", 30,  0, 1000)

gen.add("color_tolerance", int_t, 1, "tolerance for the classification based on H value", 4, 0, 255)

gen.add("min_intensity", double_t, 1, "minimum total intensity (sum of histogram values for a color hue value)", 1.0, 0.0, 5.0)

exit(gen.generate(PACKAGE, "color_classification_node", "ColorClassification"))
